Python/Product/Debugger/DebugAdapterProcess.cs

```diff
@@ -38,21 +38,29 @@ public sealed class DebugAdapterProcess : ITargetHostProcess {
         private string _interpreterOptions;
         private int _listenerPort = -1;
         private Stream _stream;
-        private Socket _socket;
-        private ManualResetEvent _connectedEvent = new ManualResetEvent(false);
 
         public DebugAdapterProcess() {
             _processGuid = Guid.NewGuid();
         }
 
+        public void Dispose() {
+            if (_stream != null) {
+                _stream.Dispose();
+            }
+            if (_process != null && !_process.HasExited) {
+                _process.Kill();
+                _process.Dispose();
```

Comment:
Should we always call `_process.Dispose()`, as long as it's not null? For example, it might have been killed in `StartProcess`

Reply:
Yes, you're right.